File: PR210-ViewModelLiveFragment/app/src/main/java/es/iessaladillo/pedrojoya/pr210/ui/detail/DetailActivityViewModel.java
package es.iessaladillo.pedrojoya.pr210.ui.detail;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

@SuppressWarnings("WeakerAccess")
public class DetailActivityViewModel extends DetailFragmentBaseViewModel {

    private final MutableLiveData<String> currentItem = new MutableLiveData<>();

    @Override
    public LiveData<String> getCurrentItem() {
        return currentItem;
    }

    @Override
    public void setCurrentItem(String item) {
        currentItem.setValue(item);
    }

}

File: PR210-ViewModelLiveFragment/app/src/main/java/es/iessaladillo/pedrojoya/pr210/ui/detail/DetailActivityViewModel.kt
package es.iessaladillo.pedrojoya.pr210.ui.detail

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

class DetailActivityViewModel : DetailFragmentBaseViewModel() {

    private val _currentItem = MutableLiveData<String>()

    override val currentItem: LiveData<String>
        get() = _currentItem

    override fun setCurrentItem(item: String) {
        _currentItem.value = item
    }

}
